cw-orch-daemon/src/tx_broadcaster.rs

```diff
@@ -0,0 +1,226 @@
+use std::time::Duration;
+
+use cosmrs::proto::cosmos::base::abci::v1beta1::TxResponse;
+use secp256k1::All;
+
+use crate::{
+    queriers::{DaemonQuerier, Node},
+    sender::Sender,
+    DaemonError, TxBuilder,
+};
+
+pub type StrategyAction =
+    fn(&mut TxBuilder, &Result<TxResponse, DaemonError>) -> Result<(), DaemonError>;
+
+pub struct RetryStrategy {
+    /// This correspond to errors caught when submitting transactions
+    pub broadcast_condition: fn(&TxResponse) -> bool,
+    /// This correspond to errors caught when simulating the transaction
+    pub simulation_condition: fn(&DaemonError) -> bool,
+    pub action: Option<StrategyAction>,
+    pub max_retries: BroadcastRetry,
+    pub current_retries: u64,
```

Comment:
`(pub crate)` probably suffices?

We should try to keep as much as possible private, that's good API practice.